### ISSUE TITLE
Error in vgg-16 architecture

### ISSUE THREAD

**Issue by AndreaPi** [keras-team/keras#9719]
Why does the [VGG-16](https://github.com/keras-team/keras/blob/master/keras/applications/vgg16.py) model lack the two final dropout layers between the fully connected layers?

**Comment by gabrieldemarmiesse** [comment]
Because the models in keras.applications are intended for inference only (or fine tuning). So things like dropout or L2 regularization are not needed.

**Comment by Sucran** [comment]
@gabrieldemarmiesse hi, thank you for your answer. If I want to fine tuning VGG-16 in a Fully Convolution Network, should I add L2 regularization to each conv layer in VGG-16 ? ( I try to add L2 regularization, but the model can not converge to the level of same version without L2 regularization )

**Comment by gabrieldemarmiesse** [comment]
I would recommend to load the vgg with notop=true, and freeze the first N layers. Instead of L2, to controle the expressiveness of the network, you can change this N value. Big N means a lot of regularization, small N means little regularization. That's my take on it. After all this, hyperparameter tuning is never an exact science, I suppose you'll never know until you try many things and see the results.

In short, you have no obligation of using L2 for fine tuning. Try many things and see what works.

**Comment by AndreaPi** [comment]
@Sucran  why not trying to add dropout after the fully connected layers, as done in the original paper?

**Comment by gabrieldemarmiesse** [comment]
You can! Try different things and see what works best.

**Comment by Sucran** [comment]
@gabrieldemarmiesse Yes, you're absolutely right. After try many many things, I solve my issue. In my case, adding L2 regularization help avoiding over-fitting, and the difference of performance is caused by optimizer ( sgd with momentum works well but adam not work, and I don't know why). Your advice really help, and I will try more different things and see what works best. 

@AndreaPi Thank you for you advise, since I working on a fully convolutional network which fine-tunes VGG16 without fully connected layers, so dropout is not in my choice (but dropout really help in some cases). It is very nice of you to pay attention to my issue, thank you.
